starfish/pipeline/filter/gaussian_low_pass.py

```diff
@@ -1,69 +1,92 @@
             Standard deviation for Gaussian kernel.
         is_volume : bool
             If True, 3d (z, y, x) volumes will be filtered, otherwise, filter 2d tiles independently.
         verbose : bool
             If True, report on the percentage completed (default = False) during processing
 
         """
+        if isinstance(sigma, tuple):
+            message = ("if passing an anisotropic kernel, the dimensionality must match the data shape ({shape}), not "
+                       "{passed_shape}")
+            if is_volume and len(sigma) != 3:
+                raise ValueError(message.format(shape=3, passed_shape=len(sigma)))
+            if not is_volume and len(sigma) != 2:
+                raise ValueError(message.format(shape=2, passed_shape=len(sigma)))
+
         self.sigma = sigma
         self.is_volume = is_volume
         self.verbose = verbose
 
     @classmethod
-    def get_algorithm_name(cls):
+    def get_algorithm_name(cls) -> str:
         return "gaussian_low_pass"
 
     @classmethod
-    def add_arguments(cls, group_parser):
+    def add_arguments(cls, group_parser: argparse.ArgumentParser) -> None:
+        group_parser.add_argument(
+            "--sigma", type=float, help="standard deviation of gaussian kernel")
         group_parser.add_argument(
-            "--sigma", default=1, type=int, help="standard deviation of gaussian kernel")
+            "--is-volume", action="store_true", help="indicates that the image stack should be filtered in 3d")
 
     @staticmethod
-    def low_pass(image, sigma) -> numpy.ndarray:
-        """Apply a Gaussian blur operation over a multi-dimensional image.
+    def low_pass(image: np.ndarray, sigma: Union[Number, Tuple[Number]]) -> np.ndarray:
+        """
+        Apply a Gaussian blur operation over a multi-dimensional image.
 
         Parameters
         ----------
-        image : np.ndarray
-            Image data
-        sigma : Union[float, int, Tuple]
+        image : np.ndarray[np.uint32]
+            2-d or 3-d image data
+        sigma : Union[Number, Tuple[Number]]
             Standard deviation of the Gaussian kernel that will be applied. If a float, an isotropic kernel will be
-            assumed, otherwise the dimensions of the kernel give (z, x, y)
+            assumed, otherwise the dimensions of the kernel give (z, y, x)
 
         Returns
         -------
         np.ndarray :
-            Blurred data in same shape as input image.
+            Blurred data in same shape as input image, converted to np.uint16 dtype.
 
         """
+        if image.dtype != np.uint16:
+            DataFormatWarning('gaussian filters only support uint16 images. Image data will be converted')
+            image = img_as_uint(image)
+
         blurred = gaussian(
             image, sigma=sigma, output=None, cval=0, multichannel=True, preserve_range=True, truncate=4.0)
 
-        blurred = blurred.astype(numpy.uint16)
+        blurred = blurred.clip(0).astype(np.uint16)
```

Comment:
The filter converts the data to float, and can produce values below zero because of how the gaussian convolution works. Those are not physiological and therefore not desirable, so we clip them.

Reply:
I should also say: simple conversion to numpy.uint16 is a bug: the data wraps around. Very bad.

Reply:
Ohhh.